Updates for Rust 1.0

brson opened the pull request:
Do not merge yet.

This removes the asterisk about laundry, adds 1.0 links to the header, and a section of stable installers to install.html, adjusts the curl flags, removes .exe in favor of .msi.

One thing to note is that instead of adding additional stable links to the header I replaced beta with stable. It felt a bit overwhelming having a third set of duplicate links, and the beta docs seemed pretty dispensable.

cc @steveklabnik

rust-highfive commented:
r? @pcwalton

(rust_highfive has picked a reviewer for you, use r? to override)

brson commented:
@steveklabnik The link to explanation of unstable features is actually not that great. It links to the books 'nightly' page which is mostly another explanation of how to download nightly. What I really wanted was an explanation of release channels and how unstable features work.

brson commented:
r? @steveklabnik

brson commented:
This updates the pitch to 'Rust is a systems programming language that runs blazingly fast, never segfaults, and guarantees thread safety.'

cc @kmcallister

steveklabnik commented on index.html:
```diff
@@ -8,22 +8,22 @@
           that runs blazingly fast,
-          prevents almost all crashes<span class="asterisk">*</span>,
-          and eliminates data races.
+          never segfaults,
```
I like changing this slogan, but 'never' is a bit strong here, given unsafe code. it's a hard caveat to fit in a slogan, though

steveklabnik commented on install.html:
```diff
@@ -5,43 +5,84 @@
+        <h3>May 15, 2015</h3>
         <p>
-	  A feature-complete preview of the 1.0.0 release.
+	  The current stable release of Rust, updated every six weeks.
```
What about something here to communicate that these updates are drop-in, easy upgrades?

brson commented:
Updated. @steveklabnik I tried to incorporate your suggestion as 'The current stable release of Rust, updated every six weeks and backwards-compatible.'

steveklabnik commented on index.html:
```diff
@@ -8,22 +8,22 @@
-          prevents almost all crashes<span class="asterisk">*</span>,
-          and eliminates data races.
+          prevents nearly all segfaults,
+          and guarantees thread safety.
```
what about 'at compile time' here, since that's the magic?


sorry for suggesting even more coats of paint :(

steveklabnik commented on install.html:
```diff
@@ -5,43 +5,84 @@
+        <h3>May 15, 2015</h3>
         <p>
-	  A feature-complete preview of the 1.0.0 release.
+	  The current stable release of Rust, updated every six weeks and backwards-compatible.
```
:heart_eyes: :+1:
